### PR TITLE
[AWS] Cleanup Naming/Typing of Boto3 resources/clients

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* It's a bit hard to follow if these are `clients` or `resources` so add typing + rename a mis-named function.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by allenyin55 — APPROVED** [review]

**Comment by ijrsvt** [comment]
Merging because failed test is failing on master:

Screen Shot 2022-06-15 at 11 56 54 AM
